### PR TITLE
Feature/iis143/replace axios

### PR DESCRIPTION
**Pull request by mi-xu**
Save 5.7 / 1.7 kB (minified / minified + gzipped) by replacing [axios](https://bundlephobia.com/result?p=axios@0.18.0) with [xmlhttprequest](https://bundlephobia.com/result?p=xmlhttprequest@1.8.0) which is also used in [uport-lite](https://github.com/uport-project/uport-lite/blob/23ed3bcc20be860328e40277f94c3d2cd43af392/package.json#L26).

Cost is that the code for handling http requests is a bit more verbose and lost the ability to cancel requests provided by `axios` (was not being used anyway).

### CODE REVIEW

File: src/register.ts [hunk 1]
```diff
@@ -1,48 +1,76 @@
...
-        Array.isArray(res.data.publicKey) && res.data.publicKey.length > 0
-      if (!docHasPublicKey) throw new Error('DID document has no public keys')
-      // validate each public key object?
-      // need to validate any other did doc attributes?
-
-      return res.data
+      response = await get(url)
     } catch (error) {
-      throw error
+      throw new Error('DID must resolve to a valid https URL')
```
Comment (mi-xu) [inline src/register.ts]: Requesting comments on the way errors are being thrown in this function. I was trying to provide informative messages describing why the did resolution failed but am not very sure of best practices here. Looking for suggestions on how to improve this.

Comment (pelle) [reply src/register.ts]: Good question. I see the reason to why you would do this, but there could be a host of different errors happening here. I can’t remember if you can include a source error in the constructor to a JavaScript error like you can in other languages.

If JS doesn’t have a good way you could maybe append the error message after the error text.

**Review by pelle — CHANGES_REQUESTED** [review]
Everything looks good. For your question about error handling, see if you can include some sort of indication of the original error for the specific one you mentioned. I think the others are quite clear.

**Comment by mi-xu** [comment]
@pelle Decided on just including the http response status in the message. Didn't end up handling all the additional error scenario handling I was saying `axios` does because they're mostly relevant for handling a variety of request configurations whereas this function is only meant to perform a single specific type of request.

**Review by pelle — APPROVED** [review]

**Review by pelle — APPROVED** [review]

**Pull request merged**